### PR TITLE
[Deconz] add support for "last_seen" channel

### PR DESCRIPTION
**Pull request by blafois**
The current binding provides support for "last_update" channel but not "last_seen". Last_update is updated when a device changes his state (battery status, or in my case of a smoke detector - when then is fire!).

Because I want to monitor if my smoke detectors are still alive, I need to check the "last_seen" value.

As last_seen is not in the WebSocket messages, it needs a refresh of the sensor:

image

### CODE REVIEW

File: bundles/org.openhab.binding.deconz/src/main/resources/ESH-INF/thing/sensor-thing-types.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
 		<channels>
 			<channel typeId="presence" id="presence"/>
 			<channel typeId="last_updated" id="last_updated"/>
+            <channel typeId="last_seen" id="last_seen"/>
```
Comment (DerOetzi) [inline bundles/org.openhab.binding.deconz/src/main/resources/ESH-INF/thing/sensor-thing-types.xml:16]: Can you please check your code formatter for correct line indention.

Comment (DerOetzi) [reply bundles/org.openhab.binding.deconz/src/main/resources/ESH-INF/thing/sensor-thing-types.xml:16]: Complete file

**Comment by blafois** [comment]
Fixed!

File: bundles/org.openhab.binding.deconz/src/main/resources/ESH-INF/thing/sensor-thing-types.xml [hunk 2]
```diff
@@ -35,6 +36,14 @@
 		<state readOnly="true" pattern="%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS"/>
 	</channel-type>
 
+	<channel-type id="last_seen">
+		<item-type>DateTime</item-type>
+    <label>Last Seen</label>
```
Comment (DerOetzi) [inline bundles/org.openhab.binding.deconz/src/main/resources/ESH-INF/thing/sensor-thing-types.xml:41]: Sorry, there is still some wrong indention

Comment (blafois) [reply bundles/org.openhab.binding.deconz/src/main/resources/ESH-INF/thing/sensor-thing-types.xml:41]: image

Atom is weird... falling back to VI then

**Comment by J-N-K** [comment]
I think there are three issue with that:

1)  The deconz binding had one great advantage over the hue binding: it does not rely on polling at all after the thing is initialized (it gets the values from websocket). This feature introduces polling for all sensors and there is now way to configure the polling interval or stop that feature.

2) The polling is not working. `scheduler.schedule` is used and that only schedules a single execution after five minutes, not a polling with an interval of five minutes.

3) Re-using the initial request `ScheduledFuture` seems to mess up initialization in some cases, there are reports in the german Facebook group of sensor things which repeatedly update themselves.

**Comment by blafois** [comment]
> I think there are two issues with that:
> 
> 1. The deconz binding had one great advantage over the hue binding: it does not rely on polling at all after the thing is initialized (it gets the values from websocket). This feature introduces polling for all sensors and there is now way to configure the polling interval or stop that feature.

Understand - but the "last seen" cannot be implemented using websocket. the information is only provided by polling the API. Only the "last change" - which for a fire sensor, does not make sense. I really need to know if I can rely on the fire sensors - and as such be able to check if they stopped reporting. 

> 2. Re-using the initial request `ScheduledFuture` seems to mess up initialization in some cases, there are reports in the german Facebook group of sensor things which repeatedly update themselves.

It works fine on my OH installation for weeks now. I can change it to declare another ScheduledFuture if you really thing it helps.

**Comment by J-N-K** [comment]
Look at the community forum or the German openHAB facebook group. Especially those with a lot of sensors seem to be affected by the new polling feature. #8193 seems to fix it.

**Pull request merged**